cmd/kyma/provision/gardener/cmd.go

```diff
@@ -142,6 +142,8 @@ func newProvider(o *Options) (*types.Provider, error) {
 		CredentialsFilePath: o.CredentialsFile,
 	}
 
+	gcpControlPlaneSet := false
```

Comment:
Instead of having this boolean and all that logic, would it not be easier, before looping through extra options just always assign `Zone[0]` and in case there is an extra config for it, it will get overwritten?
```suggestion
	if o.TargetProvider == "gcp" {
		p.CustomConfigurations["gcp_control_plane_zone"] = o.Zone[0]
	}
```